### PR TITLE
Sonobuoy run'

### LINKED ISSUE #210: Implement run subcommand -- depends on gen

### PR DESCRIPTION
**Pull request by liztio**
Same as #240, but with one commit for code and one commit for dependency updates

Closes #210

### CODE REVIEW

File: cmd/sonobuoy/app/utils/namespace/namespace.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package namespace
+
+import "github.com/spf13/cobra"
+
+type Namespace string
```
[the remark this answers is not in the document]
Comment (liztio) [reply cmd/sonobuoy/app/utils/namespace/namespace.go]: the other utils make a lot more sense as their own namespace to me (mode.Name, image.ID etc). plus I think `namespace.AddFlag()` makes more sense. But I could see arguments for flattening.

[the remark this answers is not in the document]
Comment (chuckha) [reply cmd/sonobuoy/app/utils/namespace/namespace.go]: `namespace.Namespace` is a little weird but `namespace.AddFlag` is better than `utils.AddFlag`.  I'm almost inclined to want to reorganize this into a flag package that provides these flag adding functions and then keep the types and their behavior in the utils package.

Probably not worth bike shedding over, current pattern is fine.

File: cmd/sonobuoy/app/args/image.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
+package args
+
+import (
+	"github.com/spf13/cobra"
+)
+
+// SonobuoyImage is the name of
```
Comment (chuckha) [inline cmd/sonobuoy/app/args/image.go]: accidentally some words?

File: cmd/sonobuoy/app/args/image.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+//Set the image SonobuoyImage. Returns an error when not a valid image SonobuoyImage.
+func (i *SonobuoyImage) Set(id string) error {
+	*i = SonobuoyImage(id)
+	return nil
+}
+
+// Get returns the provided SonobuoyImage, or a default if none has been provided
+func (i *SonobuoyImage) Get() string {
+	if i == nil || *i == "" {
+		return "gcr.io/heptio-images/sonobuoy:master"
```
Comment (chuckha) [inline cmd/sonobuoy/app/args/image.go]: what do you think about moving this in to https://github.com/heptio/sonobuoy/blob/master/pkg/config/config.go#L29

File: cmd/sonobuoy/app/args/mode.go [hunk 4]
```diff
@@ -7,45 +7,45 @@ import (
 	"github.com/spf13/cobra"
 )
 
-// Name is the name of the mode
+// Mode is the name of the mode
```
Comment (chuckha) [inline cmd/sonobuoy/app/args/mode.go]: stray comment?

File: cmd/sonobuoy/app/args/image.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+// SonobuoyImage is the name of
+type SonobuoyImage string
+
+// AddSonobuoyImageFlag adds a sonobuoy-image flag to existing command
+func AddSonobuoyImageFlag(id *SonobuoyImage, cmd *cobra.Command) {
+	cmd.PersistentFlags().Var(
+		id, "sonobuoy-image",
+		"what container image to use for the sonobuoy worker and container",
```
Comment (chuckha) [inline cmd/sonobuoy/app/args/image.go]: other commands have capitalized first words, don't really care if they are or not, but consistency is good

File: cmd/sonobuoy/app/args/mode.go [hunk 6]
```diff
@@ -7,45 +7,45 @@ import (
...
 )
 
-// Name identifies a specific mode
-type Name string
+// Mode identifies a specific mode
+type Mode string
 
-// Config is the
-type Config struct {
+// ModeConfig is the
```
Comment (chuckha) [inline cmd/sonobuoy/app/args/mode.go]: accidentally more words

File: cmd/sonobuoy/app/args/namespace.go [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package args
+
+import "github.com/spf13/cobra"
+
+// Namespace represents a Kubernetes namespace
+type Namespace string
+
+const usage = "the namespace to run Sonobuoy in. Only one Sonobuoy run can exist per namespace simultaneously."
```
Comment (chuckha) [inline cmd/sonobuoy/app/args/namespace.go]: capitalization

File: cmd/sonobuoy/app/operations/run.go [hunk 8]
```diff
@@ -17,21 +17,163 @@ limitations under the License.
...
+func getNames(obj runtime.Object) (string, string, error) {
+	accessor := meta.NewAccessor()
+	name, err := accessor.Name(obj)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "couldn't get name for object %T", obj)
+	}
+
+	namespace, err := accessor.Namespace(obj)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "couldn't get namespac e for object %s", name)
```
Comment (chuckha) [inline cmd/sonobuoy/app/operations/run.go]: typo in `namespac e`

**Pull request merged**